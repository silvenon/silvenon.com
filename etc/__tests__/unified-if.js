const dedent = require('dedent')
const mdx = require('@mdx-js/mdx')
const defaultLayout = require('../remark-mdx-default-layout')
const unifiedIf = require('../unified-if')

jest.mock('fs')

test('add unified plugins conditionally', async () => {
  const ifPost = (plugin) => [
    unifiedIf,
    (tree, file) => (file.path === 'custom.mdx' ? plugin : null),
  ]

  const remarkPlugins = [ifPost([defaultLayout, 'post-layout.tsx'])]

  expect(
    await mdx(
      dedent`
      # With built-in layout
    `,
      {
        filepath: 'built-in.mdx',
        remarkPlugins,
      },
    ),
  ).toMatchSnapshot()

  expect(
    await mdx(
      dedent`
      # With custom layout
    `,
      {
        filepath: 'custom.mdx',
        remarkPlugins,
      },
    ),
  ).toMatchSnapshot()
})
