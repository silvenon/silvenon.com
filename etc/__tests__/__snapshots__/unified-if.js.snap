// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add unified plugins conditionally 1`] = `
"/* @jsx mdx */


const makeShortcode = name => function MDXDefaultShortcode(props) {
  console.warn(\\"Component \\" + name + \\" was not imported, exported, or provided by MDXProvider as global scope\\")
  return <div {...props}/>
};

const layoutProps = {
  
};
const MDXLayout = \\"wrapper\\"
export default function MDXContent({
  components,
  ...props
}) {
  return <MDXLayout {...layoutProps} {...props} components={components} mdxType=\\"MDXLayout\\">
    <h1>{\`With built-in layout\`}</h1>
    </MDXLayout>;
}

;
MDXContent.isMDXComponent = true;"
`;

exports[`add unified plugins conditionally 2`] = `
"/* @jsx mdx */
import _DefaultLayout from \\"post-layout\\"

const makeShortcode = name => function MDXDefaultShortcode(props) {
  console.warn(\\"Component \\" + name + \\" was not imported, exported, or provided by MDXProvider as global scope\\")
  return <div {...props}/>
};

const layoutProps = {
  
};
const MDXLayout = _DefaultLayout
export default function MDXContent({
  components,
  ...props
}) {
  return <MDXLayout {...layoutProps} {...props} components={components} mdxType=\\"MDXLayout\\">


    <h1>{\`With custom layout\`}</h1>
    </MDXLayout>;
}

;
MDXContent.isMDXComponent = true;"
`;
